bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java

```diff
@@ -1598,9 +1598,7 @@ private ResolutionError permuteUsedBlames(ResolveSession session,
                 perm2 = allCandidates.copy();
                 permRef2.set(perm2);
             }
-            for (int reqIdx = 0; reqIdx < usedBlame.m_reqs.size(); reqIdx++)
-            {
-                Requirement req = usedBlame.m_reqs.get(reqIdx);
+            for (Requirement req : usedBlame.m_reqs) {
```

Comment:
I prefer we first make this change back at the felix project.

Reply:
Yes. Thank you for mentioning that. It was not my intention to apply it there, I just oversaw that class when reverting the clean-up for OSGi/Felix sources.

Reverted that.